### PR TITLE
Simplify to_pydatetime()

### PR DESCRIPTION
**Pull request by jbrockmendel**
Up until #17331, `Timestamp.microsecond` was very slow.  So previously it was faster to go through `convert_to_tsobject` to get a new `datetime` instance than it was to just return 
 `datetime(self.year, self.month, self.day, self.hour, self.minute, self.second, self.microsecond, self.tzinfo)`

Now it is slightly faster (and much simpler) to do it this way.

Before:

```
In [2]: ts = pd.Timestamp.now()
In [3]: %timeit ts.to_pydatetime()
The slowest run took 13.88 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 1.44 µs per loop
```

After:
```
In [2]: ts = pd.Timestamp.now()
In [3]: %timeit ts.to_pydatetime()
The slowest run took 8.51 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 1.29 µs per loop
```

With a tz-aware Timestamp the speedup is much bigger:

Before:
```
In [4]: import pytz
In [5]: ts2 = ts.replace(tzinfo=pytz.timezone('US/Pacific'))
In [6]: %timeit ts2.to_pydatetime()
The slowest run took 39.71 times longer than the fastest. This could mean that an intermediate result is being cached.
10000 loops, best of 3: 68.9 µs per loop
```

After:
```
In [7]: import pytz
In [8]: ts2 = ts.replace(tzinfo=pytz.timezone('US/Pacific'))
In [9]: %timeit ts2.to_pydatetime()
The slowest run took 8.29 times longer than the fastest. This could mean that an intermediate result is being cached.
1000000 loops, best of 3: 1.47 µs per loop
```

### CODE REVIEW

File: pandas/_libs/tslib.pyx [hunk 1]
```diff
@@ -1157,18 +1157,13 @@ cdef class _Timestamp(datetime):
...
             warnings.warn("Discarding nonzero nanoseconds in conversion",
                           UserWarning, stacklevel=2)
-        ts = convert_to_tsobject(self, self.tzinfo, None, 0, 0)
-        dts = ts.dts
-        return datetime(dts.year, dts.month, dts.day,
-                        dts.hour, dts.min, dts.sec,
-                        dts.us, ts.tzinfo)
+
+        return datetime(self.year, self.month, self.day,
+                         self.hour, self.minute, self.second,
```
Comment (chris-b1) [inline pandas/_libs/tslib.pyx]: indendation

**Comment by chris-b1** [comment]
LGTM, minus formatting nit.

**Review by jreback — CHANGES_REQUESTED** [review]
can you verify that there is an asv for this (and add one if not).

**Comment by jbrockmendel** [comment]
There is not, just added one.

**Comment by jbrockmendel** [comment]
Looks like someone added one overnight, will add to_pydatetime timing to it.

**Comment by jbrockmendel** [comment]
Is there an error message in the circleci log that I'm missing?

**Comment by jreback** [comment]
looks fine

you can click thru to circleci to see errors

**Comment by jbrockmendel** [comment]
> looks fine
>
>you can click thru to circleci to see errors

I don't ask _just_ because I want to waste your time.  I'm saying I don't see any error message in the log.

**Comment by jreback** [comment]
> I don't ask just because I want to waste your time. I'm saying I don't see any error message in the log.

I know, but some people don't actually realize you can click thru. yeah it prob timed out or something.

Wait till I merge #17619 and rebase

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks!

**Pull request merged**